pkg/xds/envoy/listeners/util.go

```diff
@@ -63,3 +70,63 @@ func UpdateFilterConfig(filterChain *envoy_listener.FilterChain, filterName stri
 func NewUnexpectedFilterConfigTypeError(actual, expected proto.Message) error {
 	return errors.Errorf("filter config has unexpected type: expected %T, got %T", expected, actual)
 }
+
+func ConvertPercentage(percentage *wrappers.DoubleValue) *envoy_type.FractionalPercent {
+	const tenThousand = 10000
+	const million = 1000000
+
+	isInteger := func(f float64) bool {
+		return math.Floor(f) == f
+	}
+
+	value := percentage.GetValue()
+	if isInteger(value) {
+		return &envoy_type.FractionalPercent{
+			Numerator:   uint32(value),
+			Denominator: envoy_type.FractionalPercent_HUNDRED,
+		}
+	}
+
+	tenThousandTimes := tenThousand * value
+	if isInteger(tenThousandTimes) {
+		return &envoy_type.FractionalPercent{
+			Numerator:   uint32(tenThousandTimes),
+			Denominator: envoy_type.FractionalPercent_TEN_THOUSAND,
+		}
+	}
+
+	return &envoy_type.FractionalPercent{
+		Numerator:   uint32(math.Round(million * value)),
+		Denominator: envoy_type.FractionalPercent_MILLION,
+	}
+}
+
+var bandwidthRegex = regexp.MustCompile(`(\d*)\s?([gmk]?bps)`)
+
+func ConvertBandwidth(bandwidth *wrappers.StringValue) (*envoy_filter_fault.FaultRateLimit_FixedLimit_, error) {
```

Comment:
This function works only for Fault Injection (`envoy_filter_fault.FaultRateLimit_FixedLimit_`) so it's not util in any way. It should be placed in fault configurer.